src/utils/types.ts

```diff
@@ -21,28 +21,52 @@ export enum TemplateType {
   LETTER = 'LETTER',
 }
 
-export type Session = {
+export type BaseTemplateFields = {
+  name: string;
+  message: string;
+};
+
+export type BaseTemplateFieldsWithSubject = BaseTemplateFields & {
+  subject: string;
+};
+
+export type Template = {
```

Comment:
You could use `Zod.inferType<$Template>` so we don't have to duplicate the typings.

Reply:
updated